src/helpers/eventCache.ts

```diff
@@ -0,0 +1,31 @@
+import { EventCache, EventDateProvider } from './EventCache';
+import { ThreadEvents } from './ThreadEvents';
+
+export class BaseEvents {
+    public threads: ThreadEvents;
+
+    private constructor(eventDateProvider?: EventDateProvider) {
+        const eventCache = new EventCache(eventDateProvider);
+        this.threads = new ThreadEvents(eventCache);
+    }
+
+    private static updateEvents(eventCache: EventCache) {
+        this.instance.threads = new ThreadEvents(eventCache);
```

Comment:
Merging with the previous cache is not something that we'd like to add now. The contents of the source and target file can be anything, it can contain Classes, Object literals, Const, Components etc..., which would make merging not feasible. In future, we will be providing an alternative method to replace duplication in files.